### DISCUSSION TITLE
How to create a query/path parameter that show up as text in the dropdown but display as an integer in the url?

### DISCUSSION THREAD

**Discussion by configureng** [fastapi/fastapi#6037] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class DropdownOption(str, Enum):
    option_1 = "Option 1"
    option_2 = "Option 2"
    option_3 = "Option 3"

@app.get("/dropdown/{option}")
async def read_dropdown(option: DropdownOption):
    option_value = option.value  
    return {"selected_option": option_value}
```

### Description

Is there someway for a dropdown in the fastapi docs to display text but have the url display the corrospoding integer value instead of the text? As you can see in the image below "Option 2" is selected and then the url displays "Option%202".  Is there anyway that instead I could have the url just display "2" by somehow mapping Option1 to 1, Option2 to 2, and Option3 to 3? Thanks.

![image](https://user-images.githubusercontent.com/111440433/220948843-6d89a42a-0704-469e-90f1-dcd248e25455.png)

### Operating System

Windows

### FastAPI Version

0.79.1

### Python Version

3.9.2

**Comment by jgould22** [comment]
I don't have an answer for you but what you want is fast api to generate a oneOf for that field sorta like what is described in this stack over flow post https://stackoverflow.com/questions/66465888/how-to-define-enum-mapping-in-openapi

I am not really sure how to coax FastAPI in generating that format though.

**Reply by configureng** [reply]
That seems like it would work but yea I have no idea how to do this in fastapi. It just seems like that this would be something that is possible. ..
